speechInteraction/modules/googleSynthesis/main.cpp

```diff
@@ -78,20 +79,24 @@ class Processing : public yarp::os::BufferedPort<yarp::os::Bottle>
     {
         this->useCallback();
         yarp::os::BufferedPort<yarp::os::Bottle >::open( "/" + moduleName + "/text:i" );
-
+        syncPort.open( "/" + moduleName + "/sync:o" );
         return true;
     }
 
     /********************************************************/
     void close()
     {
         yarp::os::BufferedPort<yarp::os::Bottle >::close();
+        syncPort.close();
     }
 
     /********************************************************/
     void onRead( yarp::os::Bottle &bot )
     {
         queryGoogleSynthesis(bot);
+        yarp::os::Bottle syncBot;
+        syncBot.addString("done");
+        syncBot.write(syncBot);
```

Comment:
@Iaxama is a standard port the one you really need here?

In pretty much all the cases a `BufferPort::write()` does a better job than `Port::write()` as the former will prevent the module from being stuck upon waiting that the receiver will have called the corresponding `read()`